c7n/resources/cw.py

```diff
@@ -833,3 +833,54 @@ def process(self, resources):
                     client.disassociate_kms_key(logGroupName=r['logGroupName'])
             except client.exceptions.ResourceNotFoundException:
                 continue
+
+
+@LogGroup.action_registry.register('put-subscription-filter')
+class SubscriptionFilter(BaseAction):
+    """Create/Update a subscription filter and associate with a log group
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: cloudwatch-put-subscription-filter
+            resource: log-group
+            actions:
+              - type: put-subscription-filter
+                filter_name: AllLambda
+                filter_pattern: ip
+                destination_arn: arn:aws:logs:us-east-1:1234567890:destination:lambda
+                distribution: Random
+    """
+    schema = type_schema(
+        'put-subscription-filter',
+        filter_name={'type': 'string'},
+        filter_pattern={'type': 'string'},
+        destination_arn={'type': 'string'},
+        distribution={'enum': ['Random', 'ByLogStream']},
+        required=['filter_name', 'destination_arn'])
+    permissions = ('logs:PutSubscriptionFilter',)
+
+    def validate(self):
+        distribution = self.data.get('distribution', '')
+        if not distribution:
```

Comment:
I think we don't want this generally required, we probably want to require it for kinesis destinations and _not_ have it for other destination types. So maybe required if `destination_arn` contains `:kinesis:`, and not allowed otherwise. And raise a `PolicyValidationError` if it's missing for kinesis destinations, or present for other destination types.

Reply:
@ajkerrigan I think it's better to not have any check here at all. The destination could be a logical destination as well. The `destination_arn` could be something like `arn:aws:logs:{region}:{account_id}:destination:r53query-log-destination`. I tested other cases like lambda function as a destination. AWS just ignores the distribution in those cases. It's safe to remove the check.

Reply:
Fair enough, my thinking was if it's going to be ignored for some destinations let's catch/prevent that earlier to avoid confusion.

Clearly that has only made things more confusing :sweat_smile: . Forget the check.